### PR TITLE
Modernize dependencies

### PR DESCRIPTION
**Pull request by defunctzombie**
### Changelog
- Update create template to React 18
- Update create template to eslint v9

### Docs
None

### Description

- Update checkout and setup-node GitHub actions
- Add `packageManager` field to `package.json` to specify that yarn is the  manager for the project (and update it to yarn v4) 
- Upgrade all prod and dev dependencies (excluding node-fetch which does not play nice with non-esm)
- Upgrade eslint to v9 for the repo
- Upgrade eslint to v9 for the generated extension

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "scripts": {
     "build": "tsc",
     "lint:ci": "eslint --report-unused-disable-directives .",
-    "lint": "eslint --report-unused-disable-directives --fix .",
+    "lint": "eslint --report-unused-disable-directives .",
```
Comment (jtbandes) [inline package.json:32]: `--report-unused-disable-directives` should be removed as this flag no longer exists (or is deprecated?).

I don't feel too strongly about `--fix` except that we have it set up with `--fix` in most of our other repos.

File: package.json [hunk 2]
```diff
@@ -39,44 +39,42 @@
...
-    "eslint-plugin-prettier": "^4.2.1",
+    "@foxglove/eslint-plugin": "2.0.0",
+    "@foxglove/tsconfig": "2.0.0",
+    "@types/jest": "29.5.14",
+    "@types/ncp": "2.0.8",
+    "@types/node": "22.9.0",
+    "@types/node-fetch": "2.6.12",
+    "@types/tmp": "0.2.6",
+    "@typescript-eslint/eslint-plugin": "8.15.0",
+    "@typescript-eslint/parser": "8.15.0",
```
Comment (jtbandes) [inline package.json:50]: unneeded

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// @ts-check
+
+const foxglove = require("@foxglove/eslint-plugin");
+const globals = require("globals");
+const tseslint = require("typescript-eslint");
```
Comment (jtbandes) [inline eslint.config.js:5]: needs to be added to package.json

File: package.json (hunk 1 above)
Comment (defunctzombie) [reply package.json:32]: I was thrown off by running lint changing my source files

File: package.json [hunk 4]
```diff
@@ -39,44 +39,42 @@
...
+    "css-loader": "7.1.2",
+    "eslint": "9.15.0",
+    "eslint-config-prettier": "9.1.0",
+    "eslint-plugin-es": "4.1.0",
+    "eslint-plugin-filenames": "1.3.2",
+    "eslint-plugin-import": "2.31.0",
+    "eslint-plugin-jest": "28.9.0",
+    "eslint-plugin-prettier": "5.2.1",
```
[the remark this answers is not in the document]
Comment (defunctzombie) [reply package.json:58]: Why don't we need them? can I also remove them from the template?

File: src/create.ts [hunk 5]
```diff
@@ -1,29 +1,29 @@
...
   "eslint-plugin-es@^4",
   "eslint-plugin-filenames@^1",
   "eslint-plugin-import@^2",
-  "eslint-plugin-jest@^27",
+  "eslint-plugin-jest@^28",
   "eslint-plugin-prettier@^5",
   "eslint-plugin-react@^7",
-  "eslint-plugin-react-hooks@^4",
+  "eslint-plugin-react-hooks@^5",
   "prettier@^3",
```
Comment (jtbandes) [inline src/create.ts:27]: seems like these should be cleaned up as well (see my comments in package.json)

File: tsconfig.json [hunk 6]
```diff
@@ -1,32 +1,15 @@
...
-    // be as strict as possible, but no stricter
-    "strict": true,
-    "noFallthroughCasesInSwitch": true,
-    "noImplicitAny": true,
-    "noImplicitReturns": true,
-    "noUncheckedIndexedAccess": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
-    "forceConsistentCasingInFileNames": true
+    "newLine": "lf"
```
Comment (jtbandes) [inline tsconfig.json:13]: I think this is in the base config already

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (jtbandes) [reply package.json:58]: They are dependencies of @foxglove/eslint-plugin now

File: tsconfig.json [hunk 7]
```diff
@@ -1,32 +1,15 @@
...
+  "extends": "@foxglove/tsconfig/base",
+  "include": ["./src/**/*"],
   "compilerOptions": {
-    "module": "commonjs",
-    "target": "es2020",
-    "lib": ["es2020"],
     "rootDir": "./src",
     "outDir": "./dist",
+    "lib": ["es2020"],
+    "target": "es2020",
```
[the remark this answers is not in the document]
Comment (defunctzombie) [reply tsconfig.json:8]: I'm concerned about this running on older node versions?

[the remark this answers is not in the document]
Comment (jtbandes) [reply tsconfig.json:8]: Perhaps the world is not ready

**Comment by defunctzombie** [comment]
This is ready for as much as I want to tackle in this PR.

I am gonna do a follow-on once I make a new release of "create-foxglove-extension" that updates the examples.

File: template/package.json [hunk 8]
```diff
@@ -11,8 +11,8 @@
   "scripts": {
     "build": "foxglove-extension build",
     "foxglove:prepublish": "foxglove-extension build --mode production",
-    "lint:ci": "eslint --report-unused-disable-directives .",
-    "lint": "eslint --report-unused-disable-directives --fix .",
+    "lint:ci": "eslint .",
+    "lint": "eslint --fix .",
```
[the remark this answers is not in the document]
Comment (defunctzombie) [reply template/package.json:15]: Do we think running lint should fix lint? I'm a bit meh on this - I think it should be more obvious that you run a command that mutates your files. I would not expect running lint to automatically do that.

[the remark this answers is not in the document]
Comment (jtbandes) [reply template/package.json:15]: I don't have strong feelings, but being consistent within a repo would be nice

**Pull request merged**